apis/v1alpha1/tlsroute_types.go

```diff
@@ -65,11 +65,28 @@ type TLSRouteStatus struct {
 
 // TLSRouteRule is the configuration for a given rule.
 type TLSRouteRule struct {
-	// Matches define conditions used for matching the rule against an
-	// incoming TLS handshake. Each match is independent, i.e. this
-	// rule will be matched if **any** one of the matches is satisfied.
-	// If unspecified, all requests from the associated gateway TLS
-	// listener will match.
+	// Matches define conditions used for matching the rule against incoming TLS
+	// connections. Each match is independent, i.e. this rule will be matched if
+	// **any** one of the matches is satisfied. If unspecified, all requests
+	// from the associated gateway listener will match.
+	//
+	// A client request may match multiple route rules. Matching precedence MUST
+	// be determined in order of the following criteria, continuing on ties:
```

Comment:
> A client request may match multiple route rules.

This can be interpreted as "it is okay for a request to match multiple route rules" but what we really want to say is that when a client matches multiple rules, use the following precedence order to break ties.

One way to put that:
Every client request should map to one route rule. In case a request matches multiple rules, the following precedence order is used to break ties:

Reply:
Good catch, this was confusing. I've merged this idea with the existing wording and copied it across all route types - hopefully the updated version is a bit easier to understand.